src/grouping-and-mapping/clients/PropertiesClient.ts

```diff
@@ -0,0 +1,125 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { AccessToken } from "@itwin/core-bentley";
+import { OperationsBase } from "../../common/OperationsBase";
+import { RequiredError } from "../../common/Errors";
+import { IPropertiesClient } from "../interfaces/IPropertiesClient";
+import { ECPropertyReference, Property, PropertyContainer, PropertyList, PropertyModify } from "../interfaces/Properties";
+import { EntityListIteratorImpl } from "../../common/iterators/EntityListIteratorImpl";
+import { Collection, getEntityCollectionPage } from "../../common/iterators/IteratorUtil";
+
+export class PropertiesClient extends OperationsBase implements IPropertiesClient {
+  private _propertiesUrl = `${this.groupingAndMappingBasePath}/datasources/imodel-mappings`;
+
+  public async createProperty(accessToken: AccessToken, mappingId: string, groupId: string, property: PropertyModify): Promise<Property> {
+    if (!this.isSimpleIdentifier(property.propertyName)) {
+      throw new RequiredError(
+        "propertyName",
+        "Field propertyName was invalid.",
+      );
+    }
+
+    if(property.ecProperties)
+      for(const ecProperty of property.ecProperties) {
+        if (!this.isValidECProperty(ecProperty)) {
+          throw new RequiredError(
+            "ecProperties",
+            "Field ecProperties was invalid.",
+          );
+        }
+      }
+
+    const url = `${this._propertiesUrl}/${encodeURIComponent(mappingId)}/groups/${encodeURIComponent(groupId)}/properties`;
+    const requestOptions: RequestInit = this.createRequest("POST", accessToken, JSON.stringify(property));
+    return (await this.fetchJSON<PropertyContainer>(url, requestOptions)).property;
+  }
+
+  public async deleteProperty(accessToken: AccessToken, mappingId: string, groupId: string, propertyId: string): Promise<Response> {
+    const url = `${this._propertiesUrl}/${encodeURIComponent(mappingId)}/groups/${encodeURIComponent(groupId)}/properties/${encodeURIComponent(propertyId)}`;
+    const requestOptions: RequestInit = this.createRequest("DELETE", accessToken);
+    return this.fetchJSON<Response>(url, requestOptions);
+  }
+
+  public async getProperty(accessToken: string, mappingId: string, groupId: string, propertyId: string): Promise<Property> {
```

Comment:
Be consistent with this type. Use `AccessToken`.

Reply:
Updated all to `AccessToken`.